docs/11-client-side-architecture.md

```diff
@@ -0,0 +1,88 @@
+- React
+- React Router
+- Redux
+- Redux Toolkit
+
+### Overview
+
+The Media Atom Maker client is a SPA React App, using Redux for state management and fetch requests to interact with the
+backend API. Users can use the client to create atoms, add metadata, upload multiple assets and choose one to be
+featured. Once published, Atoms can be featured in content in various places, but updated and managed in a single
+place. Most assets are uploaded and hosted in YouTube, using the [uploader process](../uploader/README.md).
+
+Redux manages much of the application state and is divided into slices, each managing their own state by responding to
+dispatched actions. Good examples of this are:
+
+- [Video Slice](../public/video-ui/src/slices/video.ts)
+- [Error Slice](../public/video-ui/src/slices/error.ts)
+
+When editing atom metadata, most edits are applied immediately to the Redux state. This is typically done via
+the [ManagedForm](../public/video-ui/src/components/ManagedForm/ManagedForm.jsx) components. This can present difficulty
+when save requests result in failure or edits are cancelled. In these cases there is no previous state to fall back to
+as edits have been immediately applied. The application also orchestrates saves via Thunks, which are often made up of
+several sequenced fetch requests, if parts of the process fail, it can be unclear what action users should take to
+respond.
+
+Save requests often update the entire atom model, this limits the amount users can work concurrently on atoms.
+Therefore, users rely on Presence indicators to determine whether it is safe to work on an atom. A mechanism exists to
+[report and reject potentially conflicting writes](https://github.com/guardian/media-atom-maker/blob/0d0535c5d6158a4b077c4b23c8df01790be61071/app/model/commands/UpdateAtomCommand.scala#L77-L85),
+but it doesn't work for all endpoints.
+
+## Improvements
+
+Work has been undertaken to improve the codebase in an effort to reduce the number of bugs and improve the overall UX.
+
+### Redux Toolkit
+
+The Redux constructs have been [rewritten using Redux Toolkit](https://github.com/guardian/media-atom-maker/issues/1262)
+in order to make the code more concise and the state and interactions of actions easier to reason about. Reducers and
+Actions have been converted to TypeScript as slices following
+the [RTK migration guide](https://redux.js.org/usage/migrating-to-modern-redux). To a lesser extent components have been
+updated, but this has been limited by the use of React Component Classes and JavaScript.
+
+### TypeScript Coverage
+
+The project's overall TypeScript coverage has increased significantly, roughly doubling during the course of this work.
+This has helped to identify potential bugs, improve the understanding and readability of the code, and generally make
+refactoring safer.
+
+## Future development
+
+When developing the client further, a few key areas should be considered for improvement:
+
```

Comment:
I remember RTK Query was suggested at some point.  Is it something useful for future developers to think about?

Reply:
No, not really.  I didn't look into RTK Query.  If we don't have any specific recommendations, we may just skip it.